### DISCUSSION TITLE
Frigate+ 2025.2 Base Model Update

### DISCUSSION THREAD

**Discussion by blakeblackshear** [blakeblackshear/frigate#19158] [General]
The base model in Frigate+ has now been updated to 2025.2. Once you receive an email, you should see the new base models in Frigate+. All model requests going forward will use this version. If you have a pending model request, it has automatically been updated to use 2025.2.

In this release, sampling was focused on incorporating new labels (kangaroo, skunk, school bus, canada post, and royal mail) and adding additional examples for underrepresented labels. These new labels still have relatively few examples, and performance will improve as more examples are incorporated.

I am also working to incorporate YOLOv9 as an available model type since it is supported by most hardware in the upcoming Frigate 0.16 release currently in beta. Unless I run into significant barriers, I plan to release it sometime before the next base model in the fall.

**Comment by xconverge** [comment]
Thanks, I have had a ton of success with Frigate+ so far (once I got the initial few fine tuned models in good shape). I thought I was going to burn through my 12 credits after the first 1 or 2 didn't improve too much, but after a few more trainings with more images everything stabilized and I have had really really accurate and good results!

I am eager to see if I see any urban city kangaroos I didn't know about!

**Reply by Cold-Lemonade** [reply]
@xconverge A couple of questions for you about fine tuning your models. How many images have you uploaded per camera? Did you submit the info, like camera height etc? Also, what's the resolution of your detect stream?

**Reply by xconverge** [reply]
I only have 1 camera in frigate. Its raw resolution is 4k, I downsample it to 1280x720 for detect

This shows the first 6 models I trained and the image counts, and why I was worried I was going to use up all of my credits. The first few I figured "o I can just handle a few more of these false positive and false negative cases and retrain" but really the more patient you are/the more diverse images you get especially targetting areas/things you care about most, the better. At the end of these 6 fine tuned models, I was super happy for the past 2 months and have added another ~100 or so images that will be incorporated into my updated 2025.2 based training.

<img width="1618" height="1352" alt="image" src="https://github.com/user-attachments/assets/09e84dae-5f16-4fa9-9adb-3c46570b754b" />

This means I have enough trainings left to do a retrain each time blake comes out with a new base model (during my 1 year where I can do the trainings) + 1 or 2 for maybe an architecture I dont currently need but could be nice to have for future use (yolo for example) at the end of my year.

**Reply by Cold-Lemonade** [reply]
Wow. I have 10 cameras in an urban setting. Uploading and tagging over 600 images per camera is going to take forever.

**Reply by Cold-Lemonade** [reply]
@blakeblackshear I have been using the low res stream at 704x480 resolution to run detection. So my uploaded images are also at 704x480 resolution. I recently upgraded the server running Frigate so I have more cores/threads. Should I increase the low res stream to 720p? I know that's suggested in the Frigate documentation. If I do increase to 720p, do you recommend that I start over with uploading training images? I've uploaded a bit over 1,600 images across 10 cameras so far, but I see from @xconverge that I'm going to need to upload way more before seeing any improvement. What's your advice?

**Reply by blakeblackshear** [reply]
I think you will be fine to leave the existing images in place.

The number of images needed varies a lot from user to user. Many users have good results using the base models without any fine tuning. It really depends on how similar your scenes are to the images included in the base model training. I wouldn't say that 600 images per camera is required to see good results.

With a 704x480 resolution, you should see good results for objects that are relatively close to the camera and fill up 25% or more of the frame. Smaller objects are going to be a challenge at lower resolutions. At some point you cross the threshold of what is possible to train the model to reliably recognize.

**Reply by Cold-Lemonade** [reply]
My primary goal with these cameras are to detect people approaching my house, which I assume is the same for many users. I'm in a dense urban neighborhood like Brooklyn, NY. But brownstones on my block have a rare ~14 foot setback from the sidewalk. I'm essentially monitoring an area that's 24 feet wide (width of my house) by 20 feet deep (from the cameras to the far edge of the sidewalk). I've had surprisingly good results with the default COCO model running detect at 704x480. It generally identifies people on the sidewalk in front of my house but occasionally misses them and I am fairly certain it detects most people who walk into my front garden but certainly less than 99%. (There's a complicating factor because my front garden slopes down and there's a retaining wall at the sidewalk that obscures the lower legs of most passersby.) People generally fill up much less than 25% of the frame. I am not opposed to increasing to 720p, but since I'm getting decent results at 704x480, do you think I should continue at 704x480?

**Reply by blakeblackshear** [reply]
A little hard to say since so much of it is subjective and situation specific. You can probably achieve good enough performance at that resolution. Not sure what hardware you are running on, but you may want to try the yolonas model too.

**Comment by mcmasterp** [comment]
nice! cant wait try it out and i really cant wait to see what 0.16 is like. 

one thought. how about a system where plus members can pay a bit more for more trainings? I know its not likely needed but I'd be more willy nilly requesting them and thus more likely to want to buy more if I knew it was an option. Just an idea to further support this amazing project

**Reply by xconverge** [reply]
It is an option already, right on the main page

https://frigate.video/plus/

<img width="495" height="532" alt="Screenshot 2025-07-15 at 4 43 10 PM" src="https://github.com/user-attachments/assets/1d7095f6-ec06-4a4b-b317-0a9a438bb7e0" />

**Reply by mcmasterp** [reply]
Oops I missed that. fantastic thank you

**Comment by borgqueenx** [comment]
> The base model in Frigate+ has now been updated to 2025.2. Once you receive an email, you should see the new base models in Frigate+. All model requests going forward will use this version. If you have a pending model request, it has automatically been updated to use 2025.2.
> 
> In this release, sampling was focused on incorporating new labels (kangaroo, skunk, school bus, canada post, and royal mail) and adding additional examples for underrepresented labels. These new labels still have relatively few examples, and performance will improve as more examples are incorporated.
> 
> I am also working to incorporate YOLOv9 as an available model type since it is supported by most hardware in the upcoming Frigate 0.16 release currently in beta. Unless I run into significant barriers, I plan to release it sometime before the next base model in the fall.

So a before a new base model release? Does that mean it will be the same base model, but different(better?) architecture? Did you already test it a little, what can we expect?

**Reply by blakeblackshear** [reply]
This update is a new base trained with additional images using the same architectures (yolonas and mobiledet). It is an incremental improvement for the existing labels, and adds support for the new labels mentioned.

- school_bus: works surprisingly well for having a limited number of examples in the training set, but it is also very visually distinct
- canada_post and royal_mail: they tested well on the examples I have, but they were for a limited number of users, so im not sure how well they will work in a wider range of scenarios until more examples are incorported
- kangaroo and skunk: I chose these partly because they are visually distinct. Performance on these was the lowest of the new labels, so I will definitely be adding more examples in for the next update along with additional labels.

**Reply by kirk56k** [reply]
Lucky for you a skunk just moved into my back yard. So far its come across as a cat pretty much every time, but I've been submitting images as I get them :D

**Comment by gule2004** [comment]
Hello,
I don’t fully understand the message. Does it mean that Frigate+ already supports the Hailo detector, or do we have to wait until autumn for it to be supported? Thank you.

**Reply by blakeblackshear** [reply]
Hailo is still unsupported. Since it can already run the YOLOv9 model, it should be supported once I add it. Not sure what challenges I will run into, but I expect that to be ready before the next base model on October 15th.

**Comment by dfurrer** [comment]
YOLOv9 should be a significant upgrade. Looking forward to it!

**Reply by blakeblackshear** [reply]
All the benchmarks I have seen show that it is comparable to yolonas. The main appeal is a wider range of supported hardware.

**Comment by ProsjektX** [comment]
Is there a way to review all submitted and verified images without being returned to the main image menu/overview for every image? I want to tighten up bounding boxes, make sure everything is correct, etc, but when I click on an image and hit <space>, instead of taking me to the next (verified) image I'm returned.

**Reply by blakeblackshear** [reply]
If you have a specific label you want to check, I can remove that label from the verified list for some or all of your images. Then you can progress through them like normal. Just send me an email at blake@frigate.video if that's something you want to do.

**Comment by jarzebski** [comment]
Is it possible to continue training the model with the version I started with? I spent a lot of time fine-tuning 2025.1, uploading over 1,500 photos. After updating to 2025.2, all the cameras started acting up with false detections. So I had to revert to my trained 2025.1 model, blocking any further model improvements.

**Reply by hawkeye217** [reply]
Did you request a new model or did you install 2025.2 without a new training?

**Reply by jarzebski** [reply]
without new training.

so... question is:  request new model again process my 1500+ photos uploaded before ?

**Reply by hawkeye217** [reply]
No need to re-upload anything. You can request a new model and it will be based on 2025.2, but fine-tuned with all your images. 

See the screenshot in the docs and notice how there are several fine-tuned models but they are based on two different base models: https://docs.frigate.video/plus/first_model#step-2-submit-a-model-request

**Reply by jarzebski** [reply]
oh.... great :)  I apologize for the confusion.

**Comment by JoFranke** [comment]
I dont get it - so whenever a new label is introduced and I want to use it, I need to re-verify ALL my images again? Thousand and thousand of them? That is what it shows in the camera list at least at my cameras. That does not work out. If this is really the case, I give up. Thanks.

**Reply by NickM-27** [reply]
When you add another label to a camera, the images are no longer fully verified as you need to ensure that new label is annotated / verified in all of your images. 

However, **your images are still used for training**. As long as your image has one verified label, then it is used for model training. So there is no requirement for you to go through all of the images and verify to continue training the model. 

But over time, it is necessary to re-verify the images to have the best model performance for those new labels, if the object is not annotated in older images then that will confuse the model during training. 

This is explained in a popup in the WebUI

<img width="772" height="289" alt="Screen Shot 2025-07-17 at 07 38 28 AM" src="https://github.com/user-attachments/assets/f044a945-5708-4d97-9780-4871f6a5c75a" />

**Reply by JoFranke** [reply]
Super thank you - will check it again.

**Reply by blakeblackshear** [reply]
Over time, the suggestions will become the primary tool to find any unlabeled objects in your images. This is especially true for new labels. The counts you are seeing in the UI show how many images you haven't checked for each label. It is important for the backend to know which labels each image has already been checked for by a human reviewer, so that it knows which labels the suggestion model should be scanning.

**Comment by Randym1988** [comment]
I wish there was a way to send over multiple images from frigate to plus at one time.  Unless im dumb and dont know of a more efficient way

**Reply by NickM-27** [reply]
The recommended workflow is to use the explore page with the Frigate+ and Snapshot filters enabled. From there you can use the arrow keys to move between snapshots and quickly submit the ones that make sense.

**Comment by gule2004** [comment]
I just subscribed to Frigate+ to support your work. Is it worth reporting false positive events to Frigate+ even though my Hailo is not yet supported?

**Reply by NickM-27** [reply]
I believe the past advice has been it is helpful to a point, if you spend time annotating images now it will give a good baseline for when you start running a Frigate+ model. But there is a point after ~100 or so images where you are likely better not submitting more.

**Reply by xconverge** [reply]
I was certainly glad to have some snapshots in frigate+ already from a few months prior for different weather patterns (winter vs summer), lighting conditions, houses got painted, different tree branches in the way, etc.

I agree with nick though, don't overdo it just a few "hmm I hope the new model could utilize this" or "this happens a lot and is unique to my situation so the base model might not cover it nicely"

I would not just report every false positive, because the scores you will use will be so different and so will the base model itself. There is a high chance even without a fine tuned model you would not experience the same false positives. If I have a false positive (that looks basically the same) multiple times (even across different days), I also make sure to only cover it with 1 snapshot instead of just the same-ish snapshot/bounding box multiple times. At least until I know it is incorporated into the model I am running.

**Comment by turboc1208** [comment]
you mentioned there are underrepresented items.  What are those?  If they are in my area I'll try to focus more on getting some of them.

**Comment by dopeytree** [comment]
Nice. 

Will we start to see more animal labels available?

E.g website shows Bear, raccoon, deer, etc. 

<img width="1252" height="747" alt="Screenshot 2025-07-26 at 11 22 59" src="https://github.com/user-attachments/assets/e0ece40a-07b1-477a-82e1-b2ad8f142917" />


In the UK I have cctv images of Badger, Fox & Hedgehog I would like to be able to label to help contribute.

I could train them as cats? which is the current detection.  Not really bothered what the label is just that it detects 'animals'.

<img width="1499" height="839" alt="Screenshot 2025-07-26 at 11 22 09" src="https://github.com/user-attachments/assets/23f26048-6afc-494d-8369-5c52f97a8bd3" />

**Reply by dopeytree** [reply]
Ok I can see I have to turn on extra label options in settings.

Can now do Fox. 

Be awesome to add Hedgehog & Badger 🔜

**Comment by bagobones** [comment]
Anyone seeing a slight regression in cat / dog differentiation with between 2025.1 and 2025.2 ? I feel like I have had a lot more dog matches on my cats.

The suggestion model also seems to be suggesting dog for a number of my cat more often as well.

My black/white cat has also been matching as a bird / bear / person with 2025.2, the bird and bear I get but not person.

I am all out of fine tune credits however so I am debating if I should purchase more to see if I can tune it out easily or not. I managed to get at least one fine tuning done with about 50 new images.

**Reply by NickM-27** [reply]
I saw that a little bit, mostly just cats detected as dog or bird occasionally, ran another fine tuning and it seems to be resolved.

**Comment by turboc1208** [comment]
Don’t fix it too well my husky finally quit coming out as a cat. Now if it would just stop thinking the arm of my rocking chair is my golden lab. LolSent from my iPhoneOn Aug 2, 2025, at 13:53, Nicolas Mowen ***@***.***> wrote:﻿
I saw that a little bit, mostly just cats detected as dog or bird occasionally, ran another fine tuning and it seems to be resolved.

**Comment by yousaf465** [comment]
I don't see the base models based on 2025.2 in my base models yet.

**Reply by NickM-27** [reply]
Were you an active Frigate+ subscriber when this was posted? They have been there since this was posted

**Reply by yousaf465** [reply]
<img width="725" height="538" alt="image" src="https://github.com/user-attachments/assets/f3f6296b-3016-4f8d-a05b-7819760e8cd9" />
I have been subscribed since before 2025.2

**Reply by NickM-27** [reply]
You may want to double check that your subscription is still going, remember that you only get access to base models that are released during your subscription.

After checking that, you should probably email blake@frigate.video

**Reply by yousaf465** [reply]
I just subscribed to frigate+ 2 months ago, so I think subscription is active.

```
Your Plan
Status
active
Renewal Date
```

**Reply by NickM-27** [reply]
Should email Blake then

**Reply by jonathanjsimon** [reply]
I emailed him and he replied that he found a bunch of people had missed which has now been corrected.

**Comment by gspitale** [comment]
When do we get turkey?  I have a lot of them around, and for now I'm just calling them "chicken".  :)

**Comment by patriot1889** [comment]
It was my impression that the trainings you got stayed in your account even if you were subscribed, I'm pretty sure that was the case. It appears that has changed, is that right?

**Reply by NickM-27** [reply]
What do you mean by trainings?

**Reply by patriot1889** [reply]
Apologies, The 12 model fine tunings you get when signing up.

I signed up at the start and my remaining credits stayed after my subscription lapsed, it appears they've gone now and I can't do another training. The wording has also changed on the subscription page to "up to".

**Reply by NickM-27** [reply]
@blakeblackshear can confirm but I believe that change was made early on, the trainings only last for that subscription period at which point either they will expire (if the subscription is cancelled) or they will reset to 12 (if the subscription is active).

Either way, you keep access to all models trained after the subscription ends.

**Reply by blakeblackshear** [reply]
It was implemented that way from the start. If your trainings were still there after your subscription lapsed, that must have been a bug somewhere. I had all sorts of issues with my original payment processor and the way they managed subscriptions.

Upon renewal, your trainings count is reset to 12. Unused trainings do not rollover. For those who run out and purchase additional trainings outside the subscription, those will rollover and do not expire. Don't think of your subscription as buying 12 trainings, but as access to the training features and updated base models.

**Reply by patriot1889** [reply]
Thanks for the clarification blake, clearly I was under the wrong impression. I appreciate the quick reply!
